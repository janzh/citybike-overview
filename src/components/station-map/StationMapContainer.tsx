import React from "react";

import { useStations } from "../../data/sanntid/SanntidsdataHooks";
import GeneralError from "../error/GeneralError";
import Loader from "../loader/Loader";
import StationMap from "./StationMap";

const StationMapHandler = () => {
    const { requiredDataIsLoading, requiredDataIsError } = useStations();

    if (requiredDataIsLoading) {
        return <Loader />;
    }

    if (requiredDataIsError) {
        return <GeneralError />;
    }

    return <StationMap />;
};
export default StationMapHandler;
