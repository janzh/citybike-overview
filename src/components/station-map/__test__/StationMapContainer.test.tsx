import { render } from "@testing-library/react";
import React from "react";

import { mockUseStationInformation, mockUseStationStatus } from "../../../test/mockedHooks";
import { mockResponseStationInformation, mockResponseStationStatus } from "../../../test/stationMock";
import StationMapContainer from "../StationMapContainer";

describe("StationMapContainer", () => {
    afterEach(() => {
        jest.clearAllMocks();
    });

    it("should render without failure", () => {
        mockUseStationInformation({
            data: mockResponseStationInformation,
            isError: false,
            isSuccess: true,
        });

        mockUseStationStatus({
            data: mockResponseStationStatus,
            isError: false,
            isSuccess: true,
        });

        const { asFragment } = render(<StationMapContainer />);

        expect(asFragment()).toMatchSnapshot();
    });

    describe("loading states", () => {
        it("should show loader if request to get stationInformation is not settled", () => {
            mockUseStationInformation({
                data: undefined,
                isError: false,
                isSuccess: false,
            });

            mockUseStationStatus({
                data: mockResponseStationStatus,
                isError: false,
                isSuccess: true,
            });

            const { getByText } = render(<StationMapContainer />);

            expect(getByText("Vent, siden laster")).toBeInTheDocument();
        });

        it("should show loader if request to get stationStatus is not settled", () => {
            mockUseStationInformation({
                data: mockResponseStationInformation,
                isError: false,
                isSuccess: true,
            });

            mockUseStationStatus({
                data: undefined,
                isError: false,
                isSuccess: false,
            });

            const { getByText } = render(<StationMapContainer />);

            expect(getByText("Vent, siden laster")).toBeInTheDocument();
        });
    });

    describe("error states", () => {
        it("should show loader if request to get stationInformation failed", () => {
            mockUseStationInformation({
                data: undefined,
                isError: true,
                isSuccess: false,
            });

            mockUseStationStatus({
                data: mockResponseStationStatus,
                isError: false,
                isSuccess: true,
            });

            const { getByText } = render(<StationMapContainer />);

            expect(getByText("Beklager det har skjedd en feil, prøv igjen senere.")).toBeInTheDocument();
        });

        it("should show loader if request to get stationStatus failed", () => {
            mockUseStationInformation({
                data: mockResponseStationInformation,
                isError: false,
                isSuccess: true,
            });

            mockUseStationStatus({
                data: undefined,
                isError: true,
                isSuccess: false,
            });

            const { getByText } = render(<StationMapContainer />);

            expect(getByText("Beklager det har skjedd en feil, prøv igjen senere.")).toBeInTheDocument();
        });
    });
});
