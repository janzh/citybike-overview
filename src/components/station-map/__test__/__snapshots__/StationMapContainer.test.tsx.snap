// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StationMapContainer should render without failure 1`] = `
<DocumentFragment>
  <section>
    <div
      class="leaflet-container leaflet-grab leaflet-touch-drag"
      style="position: relative;"
      tabindex="0"
    >
      <div
        class="leaflet-pane leaflet-map-pane"
        style="left: 0px; top: 0px;"
      >
        <div
          class="leaflet-pane leaflet-tile-pane"
        >
          <div
            class="leaflet-layer "
            style="z-index: 1;"
          >
            <div
              class="leaflet-tile-container leaflet-zoom-animated"
              style="z-index: 18; left: 0px; top: 0px;"
            >
              <img
                alt=""
                class="leaflet-tile"
                role="presentation"
                src="https://b.tile.openstreetmap.org/15/17362/9531.png"
                style="width: 256px; height: 256px; left: -34px; top: -174px;"
              />
            </div>
          </div>
        </div>
        <div
          class="leaflet-pane leaflet-shadow-pane"
        />
        <div
          class="leaflet-pane leaflet-overlay-pane"
        />
        <div
          class="leaflet-pane leaflet-marker-pane"
        >
          <div
            class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
            style="margin-left: 0px; margin-top: 0px; width: 40px; height: 40px; left: -1568px; top: -462px; z-index: -462;"
            tabindex="0"
          >
            
              
            <div
              class="marker"
            >
              
                
              <div
                class="marker-number "
              >
                7
              </div>
              
                
              <div
                class="marker-number "
              >
                5
              </div>
              
              
            </div>
            
       
          </div>
        </div>
        <div
          class="leaflet-pane leaflet-tooltip-pane"
        />
        <div
          class="leaflet-pane leaflet-popup-pane"
        />
      </div>
      <div
        class="leaflet-control-container"
      >
        <div
          class="leaflet-top leaflet-left"
        >
          <div
            class="leaflet-control-zoom leaflet-bar leaflet-control"
          >
            <a
              aria-label="Zoom in"
              class="leaflet-control-zoom-in"
              href="#"
              role="button"
              title="Zoom in"
            >
              +
            </a>
            <a
              aria-label="Zoom out"
              class="leaflet-control-zoom-out"
              href="#"
              role="button"
              title="Zoom out"
            >
              −
            </a>
          </div>
        </div>
        <div
          class="leaflet-top leaflet-right"
        />
        <div
          class="leaflet-bottom leaflet-left"
        />
        <div
          class="leaflet-bottom leaflet-right"
        >
          <div
            class="leaflet-control-attribution leaflet-control"
          >
            <a
              href="https://leafletjs.com"
              title="A JS library for interactive maps"
            >
              Leaflet
            </a>
             | © 
            <a
              href="https://www.openstreetmap.org/copyright"
            >
              OpenStreetMap
            </a>
             contributors
          </div>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
